cmake/sample_defs/toolchain-i686-rtems5.cmake

```diff
@@ -0,0 +1,83 @@
+
+# Note that to use this, the "RTEMS" platform module may need to be added 
+# to the system-wide CMake installation as a default CMake does not yet 
+# recognize RTEMS as a system name.  An example of this is distributed with
+# the pc-rtems PSP.
+
+# Basic cross system configuration
+set(CMAKE_SYSTEM_NAME       RTEMS)
+set(CMAKE_SYSTEM_PROCESSOR  i386)
+set(CMAKE_SYSTEM_VERSION    5)
+
+# The TOOLS and BSP are allowed to be installed in different locations.
+# If the README was followed they will both be installed under $HOME
+# By default it is assumed the BSP is installed to the same directory as the tools
+SET(RTEMS_TOOLS_PREFIX "$ENV{HOME}/rtems-${CMAKE_SYSTEM_VERSION}" CACHE PATH 
+    "RTEMS tools install directory")
+SET(RTEMS_BSP_PREFIX "${RTEMS_TOOLS_PREFIX}" CACHE PATH 
+    "RTEMS BSP install directory")
+
+# The BSP that will be used for this build
+set(RTEMS_BSP "pc686")
+
+# specify the cross compiler - adjust accord to compiler installation
+# This uses the compiler-wrapper toolchain that buildroot produces
+SET(SDKHOSTBINDIR               "${RTEMS_TOOLS_PREFIX}/bin")
+set(TARGETPREFIX                "${CMAKE_SYSTEM_PROCESSOR}-rtems${CMAKE_SYSTEM_VERSION}-")
+set(RTEMS_BSP_C_FLAGS           "-march=i686 -mtune=i686 -fno-common")
+set(RTEMS_BSP_CXX_FLAGS         ${RTEMS_BSP_C_FLAGS})
+
+
+SET(CMAKE_C_COMPILER            "${RTEMS_TOOLS_PREFIX}/bin/${TARGETPREFIX}gcc")
+SET(CMAKE_CXX_COMPILER          "${RTEMS_TOOLS_PREFIX}/bin/${TARGETPREFIX}g++")
+SET(CMAKE_LINKER                "${RTEMS_TOOLS_PREFIX}/bin/${TARGETPREFIX}ld")
+SET(CMAKE_ASM_COMPILER          "${RTEMS_TOOLS_PREFIX}/bin/${TARGETPREFIX}as")
+SET(CMAKE_STRIP                 "${RTEMS_TOOLS_PREFIX}/bin/${TARGETPREFIX}strip")
+SET(CMAKE_NM                    "${RTEMS_TOOLS_PREFIX}/bin/${TARGETPREFIX}nm")
+SET(CMAKE_AR                    "${RTEMS_TOOLS_PREFIX}/bin/${TARGETPREFIX}ar")
+SET(CMAKE_OBJDUMP               "${RTEMS_TOOLS_PREFIX}/bin/${TARGETPREFIX}objdump")
+SET(CMAKE_OBJCOPY               "${RTEMS_TOOLS_PREFIX}/bin/${TARGETPREFIX}objcopy")
+
+# Exception handling is very iffy.  These two options disable eh_frame creation.
+set(CMAKE_C_COMPILE_OPTIONS_PIC -fno-exceptions -fno-asynchronous-unwind-tables)
+
+# Link libraries needed for an RTEMS 5.x executable
+#  This was handled by the bsp_specs file in 4.11
+set(LINK_LIBRARIES              "-lrtemsdefaultconfig -lrtemsbsp -lrtemscpu")
+
+# search for programs in the build host directories
+SET(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM   NEVER)
+
+# for libraries and headers in the target directories
+SET(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY   ONLY)
+SET(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE   ONLY)
+
+SET(CMAKE_PREFIX_PATH                   /)
+
+# these settings are specific to cFE/OSAL and determines which
+# abstraction layers are built when using this toolchain
+SET(CFE_SYSTEM_PSPNAME                  pc-rtems)
+SET(OSAL_SYSTEM_BSPTYPE                 pc-rtems)
+SET(OSAL_SYSTEM_OSTYPE                  rtems)
+
```

Comment:
Note that we should only need to set `CFE_SYSTEM_PSPNAME` in the current build scripts.  The others are implied.  Could simplify here.

Reply:
See #1044